src/controllers/completion/sources/UserSource.cpp

```diff
@@ -58,7 +58,16 @@ void UserSource::initializeFromChannel(const Channel *channel)
         return;
     }
 
-    this->items_ = tc->accessChatters()->all();
+    if (getSettings()->showBroadcasterUsernameFirst)
+    {
+        this->items_ = tc->accessChatters()->allExcept(tc->getName());
+        this->items_.insert(this->items_.begin(),
+                            {tc->getName(), tc->getDisplayName()});
+    }
+    else
+    {
+        this->items_ = tc->accessChatters()->all();
+    }
```

Comment:
Would it be feasible to do the remove/move broadcaster logic in this function entirely? thus removing the `allExcept` function from ChatterSet

Reply:
I guess so. Is there any way to do it other than store a copy, erase and then insert the copy? Is this more about performance or code quality?

Reply:
More about code quality than performance yeah, the `allExcept` function sticks out a bit. It's already not going to be the most performant code, so I'd prefer it be more readable/implemented closer to where it's used

Reply:
Replied in a new comment

Reply:
The proposed solution in the new comment seems good to me :+1: